Modules/atexitmodule.c

```diff
@@ -283,11 +283,20 @@ atexit_unregister(PyObject *module, PyObject *func)
             continue;
         }
 
+        /*
+         * Increment refcounts of func and cb->func because equality check may
+         * unregister one or both
+         */
+        PyObject *cb_func = Py_NewRef(cb->func);
+        PyObject *cur_func = Py_NewRef(func);
```

Comment:
This is not needed. `func` is an argument of `atexit.unregister()`, the caller holds a reference to it.